network.tf

```diff
@@ -0,0 +1,37 @@
+resource "azurerm_subnet" "this" {
+  count = var.subnet != null ? 1 : 0
+
+  name                 = var.name
+  resource_group_name  = var.resource_group.name
+  virtual_network_name = var.subnet.virtual_network.name
+
+  enforce_private_link_endpoint_network_policies = true
+
+  address_prefixes = var.subnet.address_prefixes
+}
+
```

Comment:
It doesn't seem right to create the module in the module. You can share a subnet with private endpoint of multiple Azure resources